src/sink/tcp.rs

```diff
@@ -28,9 +28,22 @@ use crate::sink::prelude::*;
 use async_std::net::TcpStream;
 use halfbrown::HashMap;
 
+use async_tls::TlsConnector;
+use rustls::ClientConfig;
+
+use std::io::Cursor;
+use std::path::{PathBuf};
+use std::sync::Arc;
+use either::Either;
+use async_std::io::Write;
+
+type Stream = Box<dyn Write + std::marker::Unpin + Send>;
+
+
 /// An offramp streams over TCP/IP
 pub struct Tcp {
-    stream: Option<TcpStream>,
+    // TODO: check if we can/should use an enum here where we implement async_std::io::Write manually?
```

Comment:
@Licenser we had a really hard time coming up with a solution here. We need to handle both variants of streams here (`TcpStream` or `TlsStream<TcpStream>`) as if it was one, and need to store it inside this struct, without being able to use generics, as this is a dynamic thing. We resulted to a `Box<dyn SomeTrait>`. But i thought maybe you have an idea how to do it better.

Reply:
The dyn looks fine, the perf impact by dyn probably isn't noticeable given it's network IO